packages/core/src/queries/user.ts

```diff
@@ -50,6 +50,15 @@ export const createUserQueries = (pool: CommonQueryMethods) => {
       `
     );
 
+  const findUserByAddress = async (target: string, address: string) =>
+    pool.maybeOne<User>(
+      sql`
+        select ${sql.join(Object.values(fields), sql`,`)}
+        from ${table}
+        where ${fields.identities}::json#>>'{${sql.identifier([target])},address}' = ${address}
```

Comment:
❓Did we want to use this same identities json column that social uses for userInfo? Each connectors identity it grouped by `target` so it should be safe, but depends on your intentions for this field.

For some context, the metamask connector will add `{ "metamask": { "address": "0x...."} }`

Also means there is a path here for one account having multiple related blockchain addresses when using multiple connectors, which is nice. Would actually like multiple addresses for 1 user when using 1 connector too, you can have multiple accounts in metamask for example. But thats a future thing someone can contribute if its important to them